scripts/json2db.pl

```diff
@@ -189,5 +231,151 @@ sub process_alignments {
+            yes => 0,
+            no => 0,
+            absent => 0,
+            both => 0,
+        };
+        my $majority_vote = '';
+
+        if ( $motion->{motion}->{vote_events}->[0]->{counts} ) {
+            for my $count ( @{ $motion->{motion}->{vote_events}->[0]->{counts} } ) {
+                $totals->{$count->{option}} = $count->{value};
+            }
+
+            if ($totals->{yes} > $totals->{no}) {
+                $majority_vote = 'aye';
+            } else {
+                $majority_vote = 'no';
+            }
+        }
+
+        # Ignore tellers in totals
+        $totals->{yes} -= grep { $_->{option} =~ /tellaye/ } @{ $motion->{motion}->{ vote_events }->[0]->{votes} };
+        $totals->{no} -= grep { $_->{option} =~ /tellno/ } @{ $motion->{motion}->{ vote_events }->[0]->{votes} };
+
+        if ( !defined $curr_division ) {
+            my $r = $divisionadd->execute($motion_id, $house, $motion->{motion}->{text}, $yes_text, $no_text, $motion->{motion}->{date}, $motion_num, $gid, $totals->{yes}, $totals->{no}, $totals->{absent}, $totals->{both}, $majority_vote);
+            unless ( $r > 0 ) {
+                warn "problem creating division $motion_id, skipping motions\n";
+                next;
+            }
+        } elsif ( $curr_division->{division_title} ne $text ||
+                  $curr_division->{gid} ne $gid ||
+                  $curr_division->{yes_text} ne $yes_text ||
+                  $curr_division->{no_text} ne $no_text ||
+                  $curr_division->{yes_total} ne $totals->{yes} ||
+                  $curr_division->{no_total} ne $totals->{no} ||
+                  $curr_division->{absent_total} ne $totals->{absent} ||
+                  $curr_division->{both_total} ne $totals->{both} ||
+                  $curr_division->{majority_vote} ne $majority_vote
+        ) {
+            my $r = $divisionupdate->execute($gid, $text, $yes_text, $no_text, $totals->{yes}, $totals->{no}, $totals->{absent}, $totals->{both}, $majority_vote, $motion_id);
+            unless ( $r > 0 ) {
+                warn "problem updating division $motion_id from $curr_division->{division_title} to $text AND $curr_division->{gid} to $gid\n";
+            }
+        }
+
+
+        my $curr_votes = $dbh->selectall_hashref($votecheck, 'person_id', {}, $motion_id);
+
+        for my $vote ( @{ $motion->{motion}->{ vote_events }->[0]->{votes} } ) {
+            my $mp_id_num;
+            $mp_id_num = $vote->{id};
+            $mp_id_num =~ s:uk.org.publicwhip/person/::;
+            next unless $mp_id_num;
+            if ( $mp_id_num !~ /^[1-9]\d+$/ ) {
+                print "$mp_id_num doesn't look like a valid person id - skipping vote for $motion_id - " . ($dreamid || "") . "\n";
+                next;
+            }
+
+            # if we've seen this motion before then don't process it, however we want
+            # to make sure that the strong vote processing below happens so we still
```

Comment:
This comment doesn't apply as the strong vote processing is no longer present below. (Don't really know if that means you could simplify this code somehow)

Reply:
Hmm, I have no idea. Logically yes, but I don't really want to pull this bit I don't really understand apart and at worse it's just taking slightly longer.